sunpy/map/mapbase.py

```diff
@@ -1963,50 +1963,122 @@ def draw_grid(self, axes=None, grid_spacing: u.deg = 15*u.deg, annotate=True, **
                                                            rsun=self.rsun_meters,
                                                            **kwargs)
 
-    def draw_limb(self, axes=None, **kwargs):
+    def draw_limb(self, axes=None, *, resolution=1000, **kwargs):
         """
-        Draws a circle representing the solar limb
+        Draws the solar limb as seen by the map's observer.
+
+        The limb is a circle for only the simplest plots.  If the coordinate frame of
+        the limb is different from the coordinate frame of the plot axes, not only
+        may the limb not be a true circle, a portion of the limb may be hidden from
+        the observer.  In that case, the circle is divided into visible and hidden
+        segments, represented by solid and dotted lines, respectively.
 
         Parameters
         ----------
-        axes: `~matplotlib.axes` or None
-            Axes to plot limb on or None to use current axes.
+        axes : `~matplotlib.axes` or ``None``
+            Axes to plot limb on or ``None`` to use current axes.
+        resolution : `int`
+            The number of points to use to represent the limb.
 
         Returns
         -------
-        circ: list
-            A list containing the `~matplotlib.patches.Circle` object that
-            has been added to the axes.
+        visible : `~matplotlib.patches.Polygon`
+            The patch added to the axes for the visible part of the limb (i.e., the
+            "near" side of the Sun).
+        hidden : `~matplotlib.patches.Polygon`
+            The patch added to the axes for the hidden part of the limb (i.e., the
+            "far" side of the Sun).
 
         Notes
         -----
-        Keyword arguments are passed onto `matplotlib.patches.Circle`.
+        Keyword arguments are passed onto the patches.
+
+        If the limb is a true circle, ``visible`` will instead be
+        `~matplotlib.patches.Circle` and ``hidden`` will be ``None``.
+
+        To avoid trigger Matplotlib auto-scaling, these patches are added as generic
+        artists instead of patches.  One consequence is that the plot legend is not
+        populated automatically when the limb is specified with a text label.
```

Comment:
Maybe add a ref to https://matplotlib.org/stable/gallery/text_labels_and_annotations/custom_legends.html in case someone wants to add a legend for these?

Reply:
I should be able to link to that through a Sphinx reference, right?

Reply:
Done